src/particles.zig

```diff
@@ -369,91 +375,182 @@ pub const DirectionMode = union(enum(u8)) {
 	// The particle goes in a random direction
 	scatter: void,
 	// The particle goes in the specified direction
-	direction: Vec3f,
+	direction: DirectionData,
+
+	pub const DirectionData = struct {
+		dir: Vec3f,
+		radius: f32,
```

Comment:
It's hard to understand what a radius would mean here. At least in the zon it would be better to specify an opening angle instead.